Mathlib/Tactic/LibrarySearch.lean

```diff
@@ -50,6 +50,9 @@ initialize librarySearchLemmas : DeclCache (DiscrTree Name) ←
       let keys ← withReducible <| DiscrTree.mkPath type
       pure $ lemmas.insertCore keys name
 
+/-- Shortcut for calling `solveByElimImpl`. -/
+def solveByElim (depth := 6) (g : MVarId) := Lean.Tactic.solveByElimImpl false [] depth g
```

Comment:
Is there a reason this isn't using optParams? Note that they need to go at the end to work (the `depth` argument can never be elided here).

Reply:
Better, I think, to not provide a default here, only in the main `librarySearch` function, so that the default value exists in just one place.